### PR TITLE
Add Task.uploadData and update tests

### LINKED ISSUE #1655: Add uploadData(Data) case to Task enum
The inspiration for this issue is this comment https://github.com/Moya/Moya/issues/1565#issuecomment-383371379 by @ffittschen 

### Problem
He's using `.uploadFile(URL)` to upload `Data` by first writing that `Data` to a `URL`. 

### Solution
It seems reasonable to add a `.uploadData(Data)` case to the `Task` enum. 

My main concern is the scalability of the enum as we continue to add more cases. I think this is one of the drawbacks of flattening it and not using the previous strategy of composing associated values.

### PR DESCRIPTION
**Pull request by julienbodet**
Fixes #1655

### CODE REVIEW

**Comment by LucianoPAlmeida** [comment]
@sunshinejr Seems like there is an issue with the CI. Cache maybe?

File: Sources/Moya/MoyaProvider+Internal.swift [hunk 1]
```diff
@@ -195,6 +197,16 @@ private extension MoyaProvider {
         return sendAlamofireRequest(alamoRequest, target: target, callbackQueue: callbackQueue, progress: progress, completion: completion)
     }
 
+    func sendUploadData(_ target: Target, request: URLRequest, callbackQueue: DispatchQueue?, data: Data, progress: ProgressBlock? = nil, completion: @escaping Completion) -> CancellableToken {
```
Comment (LucianoPAlmeida) [inline Sources/Moya/MoyaProvider+Internal.swift:200]: If I see it correctly `sendUploadData` is exactly the same as `sendUploadFile` can we find some way to unify them in only one method? I think is ok to do it like this, but maybe would be good to have it unified :)

**Comment by sunshinejr** [comment]
@LucianoPAlmeida yeah seems like cache issue - just updated the CI, let's see how it goes!

**Comment by LucianoPAlmeida** [comment]
I think the implementation looks good, but I think it would be good to add changelog entry :) 
cc @julienbodet

**Comment by LucianoPAlmeida** [comment]
Ahh thanks for fixing the CI @sunshinejr :)

**Comment by julienbodet** [comment]
> I think the implementation looks good, but I think it would be good to add changelog entry :)
> cc @julienbodet

Good idea, it's done now!

File: Changelog.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Next
 
+### Changed
+- Add uploadData(Data) case to Task enum. [#2042](https://github.com/Moya/Moya/pull/2042) by [@julienbodet](https://github.com/julienbodet).
+
 ### Changed
 - Moya's Swift version is now Swift 5.1. [#1931](https://github.com/Moya/Moya/pull/1931) by [@BasThomas](https://github.com/BasThomas).
```
Comment (LucianoPAlmeida) [inline Changelog.md:7]: There is already a section for changed, can you move it to here?

Comment (julienbodet) [reply Changelog.md:5]: Sure!

File: Sources/Moya/MoyaProvider+Internal.swift (hunk 1 above)
Comment (julienbodet) [reply Sources/Moya/MoyaProvider+Internal.swift:200]: Good idea, but I don't see any way to do this without more refactoring. I think it's ok like this for now.

**Comment by sunshinejr** [comment]
Thanks for the PR @julienbodet, really appreciate it! We're currently trying to figure out if we can get the refactoring PR in the next release - if we do then this PR wouldn't be needed. But in case we won't, we will think about this and the other PR for improving the task type. I'll keep you posted!

**Comment by julienbodet** [comment]
> Thanks for the PR @julienbodet, really appreciate it! We're currently trying to figure out if we can get the refactoring PR in the next release - if we do then this PR wouldn't be needed. But in case we won't, we will think about this and the other PR for improving the task type. I'll keep you posted!

Great, thanks for the update!

**Pull request closed without merging**